host/entry.go

```diff
@@ -0,0 +1,293 @@
+	// output default should be returned.
+	DefaultForm() sound.Form
+
+	// DefaultSampleCodec returns the default sample codec.  In case one sample codec
+	// is not implemented or uniform accross all supported functionality, output default
+	// should be returned.
+	DefaultSampleCodec() sample.Codec
+
+	// DefaultBufSize returns the default size of the buffer provided to
+	// the caller for interactions.
+	DefaultBufSize() int
+
+	// CanOpenSource returns true if OpenSource does not return
+	// ErrUnspported.
+	CanOpenSource() bool
+
+	// OpenSource starts capturing audio.
+	//
+	// dev is the specified device, and should be nil if and only if the current
+	// entry has .DevScanner nil.  Otherwise, dev should be a device returned by
+	// the current entry .DevScan()
+	//
+	// v indicates the desired form (channels, sample rate) of the source.
+	//
+	// sco indicates the desired sample.Codec.
+	//
+	// bufSz indicates the size of buffer, in frames, which data from the
+	// hardware or software implementing the entry to sound.Source.Receive.
+	// For hardware, this is normally the size of the part of the ring buffer exposed for
+	// reading.  Implementations should use a minimal total buffer size to accomodate
+	// this constraint.
+	//
+	// OpenSource returns a triple (s, t, e) with
+	// s: sound.Source which represents captured audio.
+	// t: the start time of the first sample.
+	// e: any error
+	OpenSource(dev *libsio.Dev, v sound.Form, sco sample.Codec, bufSz int) (sound.Source, time.Time, error)
+
+	// Can open sink returns true if OpenSink does not return
+	// ErrUnsupported
+	CanOpenSink() bool
+	// OpenSink starts playing audio.
+	//
+	// dev is the specified device, and should be nil if and only if the current
+	// entry has .DevScanner nil.  Otherwise, dev should be a device returned by
+	// the current entry .DevScan()
+	//
+	// v indicates the desired form (channels, sample rate) of the source.
+	//
+	// sco indicates the desired sample.Codec.
+	//
+	// bufSz indicates the size of buffer whose data is placed
+	// in sound.Source.Receive.  This is normally the size of the part of
+	// the ring buffer exposed for playback.  Implementations should
+	// use a minimal total buffer size to safely accomodate this constraint.
+	//
+	// OpenSink returns a tuple (s, *t, e) with
+	// s: sound.Sink which represents audio for playback.
+	// t: a pointer to the start time of the first played sample, which is set
+	//    after the first successful send via the returned sound.Sink.
+	// e: any error
+	//
+	// OpenSink should
+	OpenSink(dev *libsio.Dev, v sound.Form, sco sample.Codec, bufSz int) (sound.Sink, *time.Time, error)
+
+	// CanOpenDuplex returns true if OpenDuplex does not return ErrUnsupported.
+	CanOpenDuplex() bool
+	// OpenDuplex
+	OpenDuplex(dev *libsio.Dev, iv, ov sound.Form, sco sample.Codec, bufSz int) (sound.Duplex, time.Time, error)
+
+	// ScanDevices scans devices
+	ScanDevices() []*DevScanResult
+	Devices() []*libsio.Dev
```

Comment:
Document each method of the interface, even simple ones, especially since Entry is an essential interface for other components to build upon.

Is the returned slice of devices the devices currently available on the system? Will it be automatically updated if I insert say a new microphone? Or does that require a call to `ScanDevices`?

Reply:
I had intended that scan would not be automatically updated and Notify would be the means to accomplish updating.  Will do for finishing the docs.

Reply:
new docs are there; thx.